content/guides/component-testing/quickstart-react.md

```diff
@@ -120,8 +146,256 @@ export default function Stepper({ initial = 0 }) {
+method to verify it has the correct text value.
+
+Add the following test inside the `describe` block:
+
+<code-group>
+<code-block label="src/components/Stepper.cy.jsx" active>
+
+```js
+it('stepper should default to 0', () => {
+  cy.mount(<Stepper />)
+  cy.get('span').should('have.text', '0')
+})
+```
+
+</code-block>
+</code-group>
+
+When you go back to the test runner, you should see the test pass.
+
+In the above test, we select the element by passing in "span" to `cy.get()`,
+which will select all `span` tags in our component. We only have one `span`
+currently, so this works. However, if our component evolves and we add another
+`span`, then this test could start to fail. We should use a selector that will
+be less brittle to future changes.
+
+In the `Stepper` component, the `span` tag has a `data-cy` attribute on it:
+
+```js
+<span data-cy="counter">{count}</span>
+```
+
+We can use data attributes to assign a unique id that can be used for testing
+purposes. Update the test to pass in an attribute selector to `cy.get()`:
+
+<code-group>
+<code-block label="src/components/Stepper.cy.jsx" active>
+
+```jsx
+it('stepper should default to 0', () => {
+  cy.mount(<Stepper />)
+  cy.get('[data-cy=counter]').should('have.text', '0')
+})
+```
+
+</code-block>
+</code-group>
+
+Our selector is now future-proof. For more info on writing good selectors, see
+our guide
+[Selector Best Practices](/guides/references/best-practices#Selecting-Elements).
+
+### Passing Props to Components
+
+We should also have a test to ensure the `initial` prop sets the test to
+something else besides its default value of "0". We can pass in props to the
+`Stepper` component in the `mount` method:
+
+<code-group>
+<code-block label="src/components/Stepper.cy.jsx" active>
+
+```js
+it('supports an "initial" prop to set the value', () => {
+  cy.mount(<Stepper initial={100} />)
+  cy.get('[data-cy=counter]').should('have.text', '100')
+})
+```
+
+</code-block>
+</code-group>
+
+### Testing Interactions
+
+We mounted and selected the element in the above tests but didn't interact with
+it. We should also test that the value of the counter changes when a user clicks
+the "increment" and "decrement" buttons.
+
+Add the following tests:
+
+<code-group>
+<code-block label="src/components/Stepper.cy.jsx" active>
+
+```js
+it('when the increment button is pressed, the counter is incremented', () => {
+  cy.mount(<Stepper />)
+  cy.get('[data-cy=increment]').click()
+  cy.get('[data-cy=counter]').should('have.text', '1')
+})
+
+it('when the decrement button is pressed, the counter is decremented', () => {
+  cy.mount(<Stepper />)
+  cy.get('[data-cy=decrement]').click()
+  cy.get('[data-cy=counter]').should('have.text', '-1')
+})
+```
+
+</code-block>
+</code-group>
+
+## Testing React Components with Events
+
+All the state of `<Stepper />` (ie: the count) is handled internally in the
+component. Consumers are alerted to changes to the state by passing in a
```

Comment:
Maybe we can reword the "Consumers..." bit to be more beginner friendly? 

I'm struggling to think of something succinct but even something more specific with "Consumers of the Stepper component..."

Just wary that "consumers" might not land the way the rest of the content does.